parse/parse.go

```diff
@@ -243,6 +243,11 @@ type SQLParseResult struct {
 // It requires a schema to be passed in, since SQL statements may reference
 // schema objects.
 func ParseSQL(sql string, schema *types.Schema) (res *SQLParseResult, err error) {
+	// add semicolon to the end of the statement, if it is not there
+	if sql[len(sql)-1] != ';' {
```

Comment:
I don't see anything that prevents a query from being empty from, say  `(*Service).Query`.  Best to screen out empty queries first. and/or use `strings.HasSuffix`

Reply:
Hmm, now I'm wondering what happens in other places like deployment if the `Body string` for an action or procedure does not contain valid utft. e.g. `string([]byte{random garbage that is not utf8})`.

I ask because:

```go
// string is the set of all strings of 8-bit bytes, conventionally but not
// necessarily representing UTF-8-encoded text. A string may be empty, but
// not nil. Values of string type are immutable.
type string string
```

We might be trusting that a string is valid but it doesn't have to be.  

`utf8.ValidString()` needs to be used in other contexts?  Do we test for garbage bodies in the Schema struct?

Reply:
this is a valid concern. The antlr lexer will error since the 'token' won't be recognized I think

Reply:
Just tested it, and antlr parser does complain

Reply:
I added `{
name: "non utf-8",
sql:  "\xbd\xb2\x3d\xbc\x20\xe2\x8c\x98;",
err:  nil,
},` to  Test_SQL  

The error is:
`unexpected error: detected multiple parse errors:
0: (sql) syntax error: token recognition error at: '�'
start 1:1 end 1:1
1: (sql) syntax error: token recognition error at: '�'
start 1:2 end 1:2
2: (sql) syntax error: token recognition error at: '�'
start 1:4 end 1:4
3: (sql) syntax error: token recognition error at: '⌘'
start 1:6 end 1:6
4: (sql) syntax error: mismatched input '=' expecting {'delete', 'update', 'with', 'select', 'insert'}
start 1:3 end 1:3`

Reply:
I've added Gavin's test, but I don't believe this will get parsed in the first place, so it should be ok.